Use external deps for third party libraries
Split from "deps_fix" patch from torchtext
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1ead15d46..21f85e555 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -63,5 +63,9 @@ endif()

 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${TORCH_COMPILED_WITH_CXX_ABI} -Wall ${TORCH_CXX_FLAGS}")
 
-add_subdirectory(third_party)
+find_package(double-conversion)
+find_package(re2)
+find_library(SENTENCEPIECE_LIBRARY sentencepiece PATHS $ENV{EBROOTSENTENCEPIECE}/lib64)
+find_library(SENTENCEPIECE_TRAIN_LIBRARY sentencepiece_train PATHS $ENV{EBROOTSENTENCEPIECE}/lib64)
+find_library(UTF8PROC_LIBRARY utf8proc PATHS $ENV{UTF8PROC}/lib64)
 add_subdirectory(torchtext/csrc)
diff --git a/packaging/build_wheel.sh b/packaging/build_wheel.sh
index 9882cb1b2..43e2a05d8 100755
--- a/packaging/build_wheel.sh
+++ b/packaging/build_wheel.sh
@@ -10,7 +10,6 @@ setup_env
 setup_wheel_python
 pip_install numpy future cmake>=3.18.0 ninja
 setup_pip_pytorch_version
-git submodule update --init --recursive
 python setup.py clean
 if [[ "$OSTYPE" == "msys" ]]; then
     "$script_dir/vc_env_helper.bat" python setup.py bdist_wheel
diff --git a/setup.py b/setup.py
index 080415f7e..bd84405be 100644
--- a/setup.py
+++ b/setup.py
@@ -81,7 +81,6 @@ class clean(distutils.command.clean.clean):
         # Remove build directory
         build_dirs = [
             ROOT_DIR / "build",
-            ROOT_DIR / "third_party" / "build",
         ]
         for path in build_dirs:
             if path.exists():
@@ -89,7 +88,6 @@
                 shutil.rmtree(str(path), ignore_errors=True)


-_init_submodule()
 setup_info = dict(
     # Metadata
     name="torchtext",
diff --git a/torchtext/csrc/CMakeLists.txt b/torchtext/csrc/CMakeLists.txt
index 037f814dc..658b90342 100644
--- a/torchtext/csrc/CMakeLists.txt
+++ b/torchtext/csrc/CMakeLists.txt
@@ -18,10 +18,6 @@ set(
 set(
   LIBTORCHTEXT_INCLUDE_DIRS
   ${PROJECT_SOURCE_DIR}
-  ${PROJECT_SOURCE_DIR}/third_party/sentencepiece/src
-  $<TARGET_PROPERTY:re2,INCLUDE_DIRECTORIES>
-  $<TARGET_PROPERTY:double-conversion,INCLUDE_DIRECTORIES>
-  $<TARGET_PROPERTY:utf8proc,INCLUDE_DIRECTORIES>
   ${TORCH_INSTALL_PREFIX}/include
   ${TORCH_INSTALL_PREFIX}/include/torch/csrc/api/include
   )
@@ -119,10 +115,6 @@ if (BUILD_TORCHTEXT_PYTHON_EXTENSION)
   set(
     EXTENSION_INCLUDE_DIRS
     ${PROJECT_SOURCE_DIR}
-    ${PROJECT_SOURCE_DIR}/third_party/sentencepiece/src
-    $<TARGET_PROPERTY:re2,INCLUDE_DIRECTORIES>
-    $<TARGET_PROPERTY:double-conversion,INCLUDE_DIRECTORIES>
-    $<TARGET_PROPERTY:utf8proc,INCLUDE_DIRECTORIES>
     ${TORCH_INSTALL_PREFIX}/include
     ${TORCH_INSTALL_PREFIX}/include/torch/csrc/api/include
     )
